include/range/v3/numeric/partial_sum.hpp

```diff
@@ -30,16 +30,14 @@ namespace ranges
     inline namespace v3
     {
         template<typename I, typename O, typename BOp = plus, typename P = ident,
-            typename V = value_type_t<I>,
-            typename X = concepts::Invocable::result_t<P&, V>,
-            typename Y = concepts::Invocable::result_t<BOp&, X, X>>
+            typename X = projected<projected<I, coerce<value_type_t<I>>>, P>,
+            typename Y = reference_t<X>>
         using PartialSummable = meta::strict_and<
+            SemiRegular<detail::decay_t<Y>>,
             InputIterator<I>,
-            OutputIterator<O, X>,
-            Invocable<P&, V>,
-            CopyConstructible<uncvref_t<X>>,
-            Invocable<BOp&, X, X>,
-            Assignable<uncvref_t<X>&, Y>>;
+            OutputIterator<O, detail::decay_t<Y>&>,
```

Comment:
I suppose because I would like the output sequence to depend only on the input sequence and the binary operation, and not depend on the choice of output iterator.